src/os/ui/datetime/datecontrol.js

```diff
@@ -262,9 +331,15 @@ os.ui.datetime.DateControlCtrl.prototype.updateController_ = function() {
  * @export
  */
 os.ui.datetime.DateControlCtrl.prototype.shiftDate = function(direction) {
+  var modifier = 1;
   if (!this['disabled']) {
-    this['startDate'] = os.time.offset(this['startDate'], this['duration'], direction, true);
-    this['endDate'] = os.time.offset(this['endDate'], this['duration'], direction, true);
+    if (this['duration'] === os.time.Duration.CUSTOM) {
+      // For custom durations, multiply the offset by the difference in days between the start and end dates
+      const millisecondsPerDay = 1000 * 60 * 60 * 24;
+      modifier = (this['endDate'] - this['startDate']) / millisecondsPerDay;
```

Comment:
The end date is inclusive (end of the UI's date) while start is exclusive (beginning of the UI's date), so I believe we want to add a day to the end date for this computation. I'm also not sure on the browser support for directly subtracting `Date` objects, so we may want to use `<date>.getTime()` instead to make sure this works as expected.

The result would be something like:
```
const startTime = this['startDate'].getTime();
const endTime = this['endDate'].getTime() + millisecondsPerDay;
modifier = (endTime - startTime) / millisecondsPerDay;
```
This should fix the current issue with the forward/back buttons not quite working as expected.